addons/l10n_mx/models/account_account.py

```diff
@@ -17,3 +26,23 @@ def create(self, vals_list):
             tag_id = debit_tag.id if account.code[0] in DEBIT_CODES else credit_tag.id
             account.tag_ids = [Command.link(tag_id)]
         return accounts
+
+    @api.constrains('code', 'group_id')
+    def ensure_code_conforms_coa_sat(self):
+        # filtering accounts used in Mexican reports that do not respect the regulation of the SAT (see documentation for more explanation)
+        # six first characters correspond to the group provided, a list is provided by the SAT
+        # last part is up to the user but should match XXXX_PATTERN which is detailed in the header
+        incorrect_accounts = self.filtered(
+            lambda a: a.group_id and a.account_type != 'equity_unaffected' and a.company_id.country_code == 'MX'
+            and (a.code[:6] not in SAT_CODES or not XXXX_PATTERN.fullmatch(a.code[6:]))
+        )
+        if incorrect_accounts:
+            account_names = ''.join(f'\n -   {a.name}' for a in incorrect_accounts)
```

Comment:
I think only one space is enough in `-  {account_name}`  

Nitpick, use the `\n`  in the join (and maybe add one in the following redirect